PR to Update Security Considerations document (#37)
Azure/terraform-azurerm-avm-ptn-ai-reference-implementation: security_practices.md

* **Managed Identities:** The module provisions resources with Azure Managed Identities where applicable, allowing for secure, passwordless authentication to other Azure services.
* **Role-Based Access Control (RBAC):** RBAC is enforced, ensuring that users and services have only the minimum necessary permissions to perform their tasks. You can customize role assignments based on your organizational needs.
  * **User RBAC to Access AML Workspace, Key Vault, Storage Account**
  
      | Entra Security Group           | Role                           | Purpose                |
      |--------------------------------|--------------------------------|------------------------|

>       | AKV_Secret_Admin               | key vault secret administrator | Add/Modify/Delete Secret |

Reviewer: Suggest expanding the purpose a little...

```suggestion
      | AKV_Secret_Admin               | key vault secret administrator | Add/Modify/Delete KeyVault secrets |
```

      | AML_Workspace_DS               | AzureML Data scientist         | Data Scientist Activities |

>       | AML_Workspace_ML_Operator      | AzureML Compute Operator Role  | Create Compute, Delete compute|

Reviewer: Inconsistent capitalisation of "Compute".

```suggestion
      | AML_Workspace_ML_Operator      | AzureML Compute Operator Role  | Create Compute, Delete Compute|
```

>       | User_ADLSgen_Data_contrib      | Storage Blob Data Contributor  | Read LLM Evaluation Data (Provided ADLS Gen2 is used to store evaluation responses |

Reviewer: Think there's a missing closing bracket here, and a capital P that isn't required.
```suggestion
      | User_ADLSgen_Data_contrib      | Storage Blob Data Contributor  | Read LLM Evaluation Data (provided ADLS Gen2 is used to store evaluation responses) |
```

>       | User_storage_mvnet             | Storage Blob Data Contributor  | Read/Write Data for the notebook, logs, models, snapshots, etc. and as a datastore in the Azure Storage Account created within managed-vnet |

Reviewer: Not sure I understand what you mean by "and as a datastore".
Do you mean "and access the datastore"?

* **Key Vault Integration:** Sensitive data like API keys, passwords, and connection strings can be stored in Azure Key Vault, which is integrated into the environment. Access to the Key Vault is tightly controlled using managed identities and access policies.

### 3. **Data Security**

* **Encryption:** All data at rest is encrypted using Azure Storage Service Encryption (SSE) with managed keys. You can also configure customer-managed keys (CMK) for additional control over encryption.
* **Secure Data Transfer:** Data in transit is protected using Transport Layer Security (TLS). This applies to communications between services within the environment and between the environment and external services.
*  **Data Exfiltration Prevention**
  *  Azure Machine Learning has several inbound and outbound dependencies when customers use their own VNET for provisioning AML workspace and its dependent resources. Some of these dependencies can expose a data exfiltration risk by malicious agents. AIRI will minimize data exfiltration risk by limiting inbound and outbound requirements for pattern VNet.

>   *  Resource modules created for AML Workspace and AI Studio managed virtual networks will allow restriction on egress to only approved outbound traffic, hence minimising data exfiltration risk.

Reviewer: Suggested rewording
```suggestion
  *  Resource modules created for the AML Workspace and AI Studio managed virtual networks will enforce restrictions on egress to only allow approved outbound traffic, thus minimising the data exfiltration risk.
```

  *  Code to change the access keys for Azure Storage accounts.

### 4. **Monitoring and Compliance**

* **Security Center Integration:** The module integrates with Azure Security Center to continuously monitor the security posture of the AI environment. Security Center provides recommendations and alerts for any security vulnerabilities or misconfigurations.
* **Compliance Reporting:** This reference implementation is designed to meet common compliance standards like GDPR, HIPAA, and ISO 27001. Azure Policy can be used to enforce compliance policies across all resources.

### 5. Secrets Management Inventory

 | Secret Name          | What is it?                           | Where does it live?                | How was it generated? |What's the rotation strategy?|Does it cause downtime?|What's the rotation strategy?|How does the secret get distributed to consumers?|What’s the secret’s lifespan?|
 |--------------------------------|--------------------------------|------------------------|------------------------|------------------------|------------------------|------------------|------------------|------------------|

>  |Jumpbox login          | SSH key                           | Azure Key Vault                | Code |TBP|No|TBP|TBP|TBC|

Reviewer: What is "TBP"?

Do you mean "TBD" (To be determined) or something else?

Also, what is the difference between TBP and TBC?

Author: @martinpeck thanks for picking this up! It was a typo, I have changed it to 'TBC' to be confirmed.
